src/main/java/controller/signup/SignupController.java

```diff
@@ -23,7 +24,7 @@ public SignupController(SignupInputBoundary userSignupUseCaseInteractor) {
      * @param password2 the password repeated
      */
     public void execute(String username, String userID, String password1, String password2, String email,
-                        String birthDate, String fullName, ArrayList<String> moderating, ArrayList<String> posts) {
+                        String birthDate, String fullName, List<String> moderating, List<String> posts) {
         final SignupInputData signupInputData = new SignupInputData(
```

Comment:
Ensure the SignupInputData constructor also accepts List<String> for moderating and posts to maintain type consistency